src/packages/solid/components/Input/Input.tsx

```diff
@@ -28,77 +28,108 @@ export interface InputProps extends IntrinsicNodeProps {
   actualTitle?: string;
 
   /**
-   * eyebrow of input container.
+   * index of the current cursor positions
    */
-  eyebrow?: string;
+  position?: number;
 
   /**
-   * help text for the input container.
+   * signal passed in to represent what change we want to happe in the input
    */
-  helpText?: string;
+  keyEvent: Signal<string>;
 
   /**
-   * index of the current cursor positions
+   * signal passed in to represent the actual title within the input
    */
-  position?: number;
-
-  keySignal: Signal<string>;
-
-  onRight?: KeyHandler;
-
-  onLeft?: KeyHandler;
+  titleSignal: Signal<string>;
 
   style?: Partial<InputStyles>;
 
   tone?: Tone;
 }
 
 const Input: Component<InputProps> = props => {
+  const [position, setPosition] = createSignal(props.position ?? props.actualTitle?.length ?? 0);
+
+  const formatInputText = key => {
+    if (key === undefined || key === '') {
+      return;
+    }
+
+    const inputText = props.titleSignal[0]();
+    let positions = position();
```

Comment:
can we come up with a better variable name for this? it doesn't seem like this is referencing multiple positions